pkgs/os-specific/darwin/moltenvk/default.nix

```diff
@@ -49,106 +59,113 @@ stdenv.mkDerivation (finalAttrs: {
-      -configuration Release \
-      -project MoltenVKShaderConverter.xcodeproj \
-      -scheme MoltenVKShaderConverter \
-      -arch ${stdenv.hostPlatform.darwinArch}
-    declare -A products=( [MoltenVKShaderConverter]=bin [libMoltenVKShaderConverter.a]=lib )
-    for product in "''${!products[@]}"; do
-      cp MoltenVKShaderConverter-*/Build/Products/Release/$product "$build/''${products[$product]}/$product"
-    done
-    popd
 
-    NIX_LDFLAGS+=" \
-      -lobjc \
-      -lMoltenVKShaderConverter \
-      -lspirv-cross-reflect"
-
-    pushd MoltenVK
     xcodebuild build \
+      SYMROOT=$PWD/Products OBJROOT=$PWD/Intermedates \
       -jobs $NIX_BUILD_CORES \
       -configuration Release \
-      -project MoltenVK.xcodeproj \
-      -scheme MoltenVK-macOS \
+      -project MoltenVKPackaging.xcodeproj \
+      -scheme 'MoltenVK Package (macOS only)' \
+      -destination generic/platform=macOS \
       -arch ${stdenv.hostPlatform.darwinArch}
-    cp MoltenVK-*/Build/Products/Release/dynamic/libMoltenVK.dylib "$build/lib/libMoltenVK.dylib"
-    popd
-  '';
-
-  installPhase = ''
-    mkdir -p "$out/lib" "$out/share/vulkan/icd.d" "$bin/bin" "$dev/include/MoltenVK"
-    cp build/bin/MoltenVKShaderConverter "$bin/bin/"
-    cp build/lib/libMoltenVK.dylib "$out/lib/"
-    cp MoltenVK/MoltenVK/API/* "$dev/include/MoltenVK"
-    install -m644 MoltenVK/icd/MoltenVK_icd.json "$out/share/vulkan/icd.d/MoltenVK_icd.json"
-    substituteInPlace $out/share/vulkan/icd.d/MoltenVK_icd.json \
-      --replace-fail ./libMoltenVK.dylib "$out/lib/libMoltenVK.dylib"
   '';
 
-  postFixup = ''
-    install_name_tool -id "$out/lib/libMoltenVK.dylib" "$out/lib/libMoltenVK.dylib"
-    codesign -s - -f "$out/lib/libMoltenVK.dylib"
-    codesign -s - -f "$bin/bin/MoltenVKShaderConverter"
-  '';
+  installPhase =
+    ''
+      mkdir -p "$out/lib" "$out/share/vulkan/icd.d" "$bin/bin" "$dev"
+
+      cp Package/Release/MoltenVKShaderConverter/Tools/MoltenVKShaderConverter "$bin/bin"
+      cp -r Package/Release/MoltenVK/include "$dev"
+    ''
+    + lib.optionalString isStatic ''
+      cp Products/Release/libMoltenVK.a "$out/lib"
+    ''
+    + lib.optionalString (!isStatic) ''
+      # MoltenVK’s Xcode project builds the dylib, but it doesn’t seem to work with xcbuild.
+      # This is based on the script versions prior to 1.2.8 used to create the dylib.
+      clang++ -Wl,-all_load -Wl,-w \
+        -dynamiclib \
+        -compatibility_version 1.0.0 -current_version 1.0.0 \
+        -LProducts/Release \
+        -framework AppKit \
+        -framework CoreGraphics \
+        -framework Foundation \
+        -framework IOKit \
+        -framework IOSurface \
+        -framework Metal \
+        -framework QuartzCore \
+        -lobjc \
+        -lMoltenVKShaderConverter \
+        -lspirv-cross-reflect \
+        -install_name "$out/lib/libMoltenVK.dylib" \
+        -o "$out/lib/libMoltenVK.dylib" \
+        -force_load Products/Release/libMoltenVK.a
```

Comment:
this clang++ seems more apt for the postBuild hook. write it into Products/Release/libMoltenVK.dylib next to the.a

the installPhase and the buildPhase both need the runHook preBuild/postBuild.

Reply:
This should be done. I tried to follow where upstream puts the dylib in the `Package` hierarchy. I didn’t create a fake xcframework for the static library though.